roadmap/c-unwind-abi.md

```diff
@@ -0,0 +1,38 @@
+# "C unwind" ABI
+
+## Summary
+
+Functions that use the plain `"C"` ABI are **not permitted to unwind**.
+Doing so is "undefined behavior", which means that the compiler is free
+to assume it cannot happen. The results are therefore unpredictable.
+It is always a bug.
+
+We are adding therefore a new ABI `"C unwind"`. This ABI can be used
+to indicate functions that follow the C ABI and which **may** unwind.
+The precise unwinding mechanism that such functions use will depend on
+the target definition. In general, though, it is meant to match the
+"native" unwinding mechanism of the target: typically whatever C++
+compilers use.
+
+**Warning:** We are still in the process of fully specifying when and
+how "unwinding interop" works between native functions and Rust. This
+roadmap item alone **only** adds the ABI -- it does not define what
+should happen if unwinding actually occurs, and hence **even if you
+use this ABI**, unwinding across a `"C unwind"` ABI barrier is **still
+undefined behavior**. However, it is our **intent** to define that
```

Comment:
Hm, do we want to call this "undefined behavior", without qualification? I had thought we intended (even in this initial minimal step) to at least specify enough of the behavior to prohibit `rustc` from emitting `nounwind` for the new ABI. As written, it would be an entirely valid implementation to treat `"C unwind"` as an alias for `"C"`.

Reply:
I say it is undefined behavior, full stop. It just happens to be behavior we **intend** to define. But at the moment, it is not defined. It could do anything (including what it actually does).

Reply:
Hm, the oddity to me is that this is in a section that assumes the existence of `"C unwind"`, i.e., it is already defining new semantics. I'm not sure it's beneficial to introduce new semantics with no behavioral guarantees whatsoever.